Release 6.5.1 (#1197)
aio-libs/multidict: CHANGES.rst

.. towncrier release notes start

6.5.1
=====

*(2025-06-24)*

Bug fixes
---------

- Fixed a bug in C implementation when multidict is resized and it has deleted slots.

  The bug was introduced by multidict 6.5.0 release.

  Patch by :user:`asvetlov`.

  *Related issues and pull requests on GitHub:*
  :issue:`1195`.

Contributor-facing changes
--------------------------

- A pair of code formatters for Python and C have been configured in the pre-commit tool.

  *Related issues and pull requests on GitHub:*
  :issue:`1123`.

- Shorted fixture parametrization ids.

  For example, ``test_keys_view_xor[case-insensitive-pure-python-module]`` becomes ``test_keys_view_xor[ci-py]`` -- by :user:`asvetlov`.

  *Related issues and pull requests on GitHub:*
  :issue:`1192`.

- The :file:`reusable-cibuildwheel.yml` workflow has been refactored to
  be more generic and :file:`ci-cd.yml` now holds all the configuration
  toggles -- by :user:`webknjaz`.

  *Related issues and pull requests on GitHub:*
  :issue:`1193`.

----

> 6.5.0

Reviewer: Could you add a `.. note::` under the date below saying that it's yanked?

.. towncrier release notes start

6.5.1
=====

*(2025-06-24)*

Bug fixes
---------

> - Fix a bug in C implementation when during multidict resizing the function could drop

Author: ```suggestion
- Fixed a bug in C implementation when during multidict resizing the function could drop
```

>   last item under some circumstances.

Author: It occurs when the dict has deleted items and it should be resized to add a new key/value pair.
I'm afraid, for the end user, it really looks like a mysterious circumstance, sometimes all works fine, but in other cases it doesn't.
It all depends on the actual internal state, which is completely hidden from the user. I think that my trial to describe the exact state for error reproducing doesn't help with understanding. Curious people could open the corresponding PR and see the change though, the diff has only 4 lines of code except a test.

>   last item under some circumstances.

Reviewer: Still, write something more specific instead of building a suspense. It's not a true crime novel. The changelog should not contain references nobody knows how to understand.

> - Fixed a bug in C implementation when during multidict resizing the function could drop
>   last item under some circumstances.

Author: ```suggestion
- Fixed a bug in C implementation when multidict is resized and it has deleted slots.
```

Author: @webknjaz is it better?

Reviewer: Yes

  The bug was introduced by multidict 6.5.0 release.

  Patch by :user:`asvetlov`.

  *Related issues and pull requests on GitHub:*
  :issue:`1195`.

Contributor-facing changes
--------------------------

> - Setup pre-commit hook for auto-formatting both Python and C code.

Reviewer: ```suggestion
- A pair of code formatters for Python and C have been configured in the pre-commit tool.
```

Author: As you wish. 
Technically 'setup' verb has the same spelling in both present and past forms, isn't it?

Reviewer: No, it's not. `set up` is a verb or an adjective. `setup` or `set-up` is a noun.
